Promise-ify nreplClient.ts.

mhansen opened the pull request:
Convert a few callers who had to wrap the callbacks in Promises to use the .then() Promise-transforming APIs instead.

Removed a duplicate definition of 'document' in clojureSuggest.ts which was flagged as a warning after this refactoring.

fasfsfgs commented on src/nreplClient.ts:
```diff
@@ -42,31 +42,39 @@ interface nREPLCloseMessage {
-const stacktrace = (session: string): Promise<any> => send({ op: 'stacktrace', session: session });
+const stacktrace = (session: string): Promise<any> => {
+    return send({ op: 'stacktrace', session: session });
+}
```
Is there any reason to chance this @mhansen?
[Because they mean exactly the same.](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
Do you think it's easier to read or something like that?

mhansen replied:
I just changed it to be consistent with the other lines in this file, where a bunch of the other send functions have the body of the function in a block on a new line.

But I don't feel strongly about this change at all. Actually, looking back on the original state, it seems like there is some consistency that I missed in the first place - that if it's an easy one-liner, it's done as a one-liner style, but if it wouldn't fit on one line, it's done on two lines. So maybe this was unnecessary :-)

fasfsfgs replied:
That's ok. =) It's up to you (and @avli). I'll trust whatever you think it's best then.
If you think the old way was better, just commit those changes back. If not, that's fine as well.

mhansen replied:
Reverted these bits.

fasfsfgs commented:
Thank you very much for your work @mhansen!
That's a great and needed refactoring!
Can you check my review please?

mhansen commented:
Sorry, I'm new and I think I'm probably missing something - how do I check your review? I don't see any comments on the diffs here: https://github.com/avli/clojureVSCode/pull/49/files?diff=split. Is there a trick to show review comments?

mhansen commented:
Ah, I see what you mean: I left some comments on your pull request #50. Looks good!

avli commented:
@fasfsfgs @mhansen Guys, could you explain me what review are you talking about. I don't see any comments...

fasfsfgs commented:
@avli @mhansen That was my bad. I commented part of @mhansen's code but I didn't submit it. I just did now. And it's nothing serious or anything like that.
The work here is great!
I'm just curious about one particular change.

mhansen committed:
Promise-ify nreplClient.ts.

Convert a few callers who had to wrap the callbacks in Promises to use the .then() Promise-transforming APIs instead.

Removed a duplicate definition of 'document' in clojureSuggest.ts which was flagged as a warning after this refactoring.

fasfsfgs reviewed (approved):
Great work imo. Thank you @mhansen.

avli commented:
Great job indeed, guys. Thank you very much!
